package.json

```diff
@@ -109,12 +109,36 @@
                 "when": "suggestWidgetVisible && textInputFocus",
                 "intellij": "Choose Lookup Item Replace"
             },
+            {
+                "key": "ctrl+shift+enter",
+                "mac": "cmd+shift+enter",
+                "command": "runCommands",
+                "args": {
+                    "commands": [
+                        "acceptSelectedSuggestion",
+                        {
+                            "command": "cursorMove",
+                            "args": {
+                                "to": "wrappedLineEnd"
+                            }
+                        },
+                        {
+                            "command": "editor.action.insertSnippet",
+                            "args": {
+                                "snippet": ";"
+                            }
+                        }
+                    ]
+                },
+                "when": "suggestWidgetVisible && textInputFocus && (resourceLangId == 'javascript' || resourceLangId == 'typescript') && config.javascript.format.semicolons == 'insert'",
```

Comment:
nit: editorLangId == 'typescript' is better than resourceLangId since there is no need the explorer value
https://code.visualstudio.com/api/references/when-clause-contexts#available-context-keys